hassio/src/hassio-main.ts

```diff
@@ -56,6 +60,30 @@ export class HassioMain extends SupervisorBaseElement {
     });
 
     makeDialogManager(this, this.shadowRoot!);
+
+    if (atLeastVersion(this.hass.config.version, 2021, 2, 4)) {
+      this.hass.connection.subscribeEvents(
+        (event) =>
+          this._handleSupervisorEvent((event as any).data as SupervisorEvent),
+        "supervisor_event"
+      );
+    }
+  }
+
+  private _handleSupervisorEvent(event: SupervisorEvent): void {
+    if (
+      event.event === "supervisor-update" &&
+      event.update_key !== undefined &&
+      event.data !== undefined &&
+      this.supervisor !== undefined
+    ) {
+      const data: Partial<Supervisor> = {};
+      data[event.update_key] = {
+        ...this.supervisor![event.update_key],
+        ...event.data,
```

Comment:
Does this mean a key can never be removed in an update? Doesn't the update contain the entire new object?

Reply:
a key should never be removed.

The idea here is that we can add partial updates in the backend for certain things, like the core state.
Instead of generating the entire JSON data and sending that over to core (which will populate the DB since it's an event), we could just send the new core state